react/components/LazyRender.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { FunctionComponent, useRef, useState } from 'react'
+import { useOnView } from '../hooks/viewDetection'
+
+interface Props {
+  height?: number
+  offset?: number
+  debug?: boolean
+}
+
+const LazyRender: FunctionComponent<Props> = ({
+  children,
+  height = 400,
+  offset = 300,
+  debug = false,
+}) => {
+  const ref = useRef(null)
+  const [hasBeenViewed, setHasBeenViewed] = useState(false)
+
+  useOnView({
+    ref,
+    onView: () => {
+      setHasBeenViewed(true)
+      if (debug) {
```

Comment:
Are we going to leave this log here? Could we add a `process.env.NODE_ENV` check at least?